src/components/search/SearchDirective.js

```diff
@@ -149,40 +153,31 @@ goog.require('ga_search_type_directives');
             // Standard query then
             var url = gaUrlUtils.append($scope.options.searchUrl,
                                         'searchText=' + encodeURIComponent(q));
-            url = gaUrlUtils.append(url, 'lang=' + $translate.use());
-            url = $scope.options.applyTopicToUrl(url,
-                $scope.childoptions.currentTopic);
+            url = gaUrlUtils.append(url, 'lang=' + gaLang.get());
+            url = url.replace('{Topic}', gaTopic.get().id);
 
             $scope.childoptions.baseUrl = url;
             $scope.childoptions.query = q;
           }
         } else {
           blockQuery = false;
+          $scope.$broadcast('gaSearchReset');
```

Comment:
Using `$scope.childoptions.query = ''` here would have the same effect and we would not need `gaSearchReset` event at all (we do this in the clearInput function already).